Mathlib/Order/Directed.lean

```diff
@@ -373,6 +373,14 @@ lemma prod {d₁ : Set α} {d₂ : Set β} (h₁ : DirectedOn (· ≼₁ ·) d
   obtain ⟨r₂, hdr₂, hpr₂, hqr₂⟩ := h₂ _ hpd.2 _ hqd.2
   exact ⟨⟨r₁, r₂⟩, ⟨hdr₁, hdr₂⟩, ⟨hpr₁, hpr₂⟩, ⟨hqr₁, hqr₂⟩⟩
 
+lemma prod_all_dominated {d : Set (α × β)} (hd : DirectedOn (fun p q ↦ p.1 ≼₁ q.1 ∧ p.2 ≼₂ q.2) d) :
```

Comment:
I think this name will never be found by anyone. Can you come up with some alternatives?

Reply:
@YaelDillies it feels like there ought to be a name for the situation where `s ⊆ t` and for all `x∈t` there exists `y∈s` such that `x ≤ y`, but I don't know what that is?

Reply:
It's called "dominated" in the informal literature, but mathlib doesn't have a definition because it's also just `t \subseteq upperSet s`.

Maybe mathlib should eventually acquire a definition (eg transitivity is painful to prove inline)

Reply:
How about something like:
```
lemma DirectedOn.fst_image_times_snd_image_subset_lowerClosure {d : Set (α × β)}
    (hd : DirectedOn (· ≤ ·) d) : (Prod.fst '' d) ×ˢ (Prod.snd '' d) ⊆ lowerClosure d
```
This does require pulling in `Mathlib.Order.UpperLower.Closure` as an additional import though.